### PR TITLE
Add errortype for spain soap

### PR DESCRIPTION
**Pull request by ivanliavonchykmews**
## Description

Extends the Spainsh error result with Error type field requireq for https://mews.myjetbrains.com/youtrack/issue/kanban-398

## Type of change
- [ ] Bug fix.
- [x] Feature.
- [ ] Consolidation.

## Checklist
- [ ] Is breaking change.
- [ ] Documentation updated.
- [ ] Tests included (please specify cases).

### CODE REVIEW

File: src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs [hunk 1]
```diff
@@ -8,12 +9,20 @@ private ErrorResult(string code, string message)
...
 
+        public ErrorType ErrorType { get; }
         public string Message { get; }
 
         public static ErrorResult Create(string code, string message) => new ErrorResult(code, message);
+
+        private ErrorType ConvertErrorType(string message)
+        {
+            return message.Match(
+                "Codigo[401].Certificado revocado", _ => ErrorType.CertificateRevoked);
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs:25]:
```suggestion
                "Codigo[401].Certificado revocado", _ => ErrorType.CertificateRevoked,
                _ => ErrorType.Unknown
       )
```

Comment (abdallahbeshi) [reply src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs:25]: I'm sure at some point we will extend this, but for now let's make sure we return Unknown error when we're not sure what it is.

File: src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs [hunk 2]
```diff
@@ -8,12 +9,20 @@ private ErrorResult(string code, string message)
...
         }
 
         public string Code { get; }
 
+        public ErrorType ErrorType { get; }
         public string Message { get; }
 
         public static ErrorResult Create(string code, string message) => new ErrorResult(code, message);
+
+        private ErrorType ConvertErrorType(string message)
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs:22]:
```suggestion
        private ErrorType MapErrorType(string message)
```

File: src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs [hunk 3]
```diff
@@ -19,10 +20,11 @@ private ErrorResult(string code, string message)
 
         public static ErrorResult Create(string code, string message) => new ErrorResult(code, message);
 
-        private ErrorType ConvertErrorType(string message)
+        private ErrorType MapErrorType(string message)
         {
             return message.Match(
-                "Codigo[401].Certificado revocado", _ => ErrorType.CertificateRevoked);
+                "Codigo[401].Certificado revocado", _ => ErrorType.CertificateRevoked,
+                _ => ErrorType.Unknown);
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs:27]: Move the closing parentheses to the next line and align it with `return message.Match(`

File: src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using FuncSharp;
+using Mews.Fiscalizations.Spain.Model.Enums;
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs:3]: removed unused using.

Comment (ivanliavonchykmews) [reply src/Spain/Mews.Fiscalizations.Spain/Model/Response/ErrorResult.cs:3]: Strange it shoould be a folder inside the same project...Oh I just didn't save the file before commit. my bad

**Review by macous — APPROVED** [review]

**Review by abdallahbeshi — APPROVED** [review]

**Comment by abdallahbeshi** [comment]
cc @ivanliavonchykmews https://github.com/MewsSystems/fiscalizations/pull/114/commits/3addcb72a8a3dd22ef86357cddbfdc5c640858ab

**Comment by ivanliavonchykmews** [comment]
> cc @ivanliavonchykmews [3addcb7](https://github.com/MewsSystems/fiscalizations/commit/3addcb72a8a3dd22ef86357cddbfdc5c640858ab)

nice one

**Review by KaliCZ — APPROVED** [review]

**Pull request merged**